### ISSUE TITLE
Istioctl Egress gateway install option doesn't work

### ISSUE THREAD

**Issue by balamuru** [istio/istio#20760]
**Bug description**
Egress gateway doesn't install when the "set" flag option is used as per documented instructions

[ ] Configuration Infrastructure
[ ] Docs
[ X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Egress gateway svc and pod are available
**Steps to reproduce the bug**
* As per the following instructions , attempt to install istio with the egress gateway enabled https://preliminary.istio.io/docs/tasks/traffic-management/egress/egress-gateway/#deploy-istio-egress-gateway
```
istioctl manifest apply --set values.global.istioNamespace=istio-system \
    --set values.gateways.istio-ingressgateway.enabled=false \
    --set values.gateways.istio-egressgateway.enabled=true

```
* The egress gateway doesn't deploy
```
$  kubectl get pod  -n istio-system
NAME                                      READY   STATUS    RESTARTS   AGE
istio-citadel-f78ff689-4b6jz              1/1     Running   0          14m
istio-galley-fd9694b9c-bk4d8              2/2     Running   0          14m
istio-ingressgateway-576f494c7c-jl4wk     1/1     Running   0          14m
istio-nodeagent-kklc8                     1/1     Running   0          32m
istio-nodeagent-wgnxw                     1/1     Running   0          32m
istio-pilot-5c68bb85df-pd88c              2/2     Running   0          14m
istio-policy-587798bdcc-5dh9z             2/2     Running   0          14m
istio-sidecar-injector-7dd87d7989-mcjlx   1/1     Running   1          3d23h
istio-telemetry-75c64d988b-brcsh          2/2     Running   0          14m
prometheus-66c5887c86-kctvj               1/1     Running   1          3d23h


kc get svc -n istio-system
NAME                     TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                                                                                      AGE
istio-citadel            ClusterIP      10.96.132.244   <none>        8060/TCP,15014/TCP                                                                                                           3d23h
istio-galley             ClusterIP      10.96.199.32    <none>        443/TCP,15014/TCP,9901/TCP,15019/TCP                                                                                         3d23h
istio-ingressgateway     LoadBalancer   10.96.88.186    <pending>     15020:31532/TCP,80:32368/TCP,443:31918/TCP,15029:32594/TCP,15030:31691/TCP,15031:31880/TCP,15032:30829/TCP,15443:30617/TCP   3d23h
istio-pilot              ClusterIP      10.96.219.78    <none>        15010/TCP,15011/TCP,8080/TCP,15014/TCP                                                                                       3d23h
istio-policy             ClusterIP      10.96.121.163   <none>        9091/TCP,15004/TCP,15014/TCP                                                                                                 3d23h
istio-sidecar-injector   ClusterIP      10.96.74.155    <none>        443/TCP                                                                                                                      3d23h
istio-telemetry          ClusterIP      10.96.177.130   <none>        9091/TCP,15004/TCP,15014/TCP,42422/TCP                                                                                       3d23h
prometheus               ClusterIP      10.96.209.21    <none>        9090/TCP      
```
**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```
istioctl version --remote
client version: 1.4.3
citadel version: 1.4.3
galley version: 1.4.3
ingressgateway version: 1.4.3
nodeagent version: 
nodeagent version: 
pilot version: 1.4.3
policy version: 1.4.3
sidecar-injector version: 1.4.3
telemetry version: 1.4.3
data plane version: 1.4.3 (11 proxies)

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-10T03:03:57Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:12:17Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}

```
**How was Istio installed?**
istioctl
**Environment where bug was observed (cloud vendor, OS, etc)**
upstream kubernetes

**Comment by blhagadorn** [comment]
This is definitely a bug, `istioctl manifest generate --set values.global.istioNamespace=istio-system \
    --set values.gateways.istio-ingressgateway.enabled=false \
    --set values.gateways.istio-egressgateway.enabled=true | grep EgressGateway` --> `# EgressGateway component is disabled.` Also interestingly your IngressGateway was generated

**Comment by balamuru** [comment]
I'm think the issue  is related to this https://github.com/istio/istio/issues/19799 . It hasn't been fixed in 1.4.5 . I'm guessing the fix is projected for 1.5

**Comment by Mistic92** [comment]
Any other way to manual deploy egress gateway?

**Comment by balamuru** [comment]
> Any other way to manual deploy egress gateway?

* Check if it works in Istio 1.5 if that is a possibility.
* You could use the following istioctl install options in a pinch, although it isn't recommended ( https://github.com/istio/istio/issues/19799#issue-542609873 )

```
..
--set values.gateways.istio-egressgateway.enabled=true \
--set gateways.components.egressGateway.enabled=true \
..
```

**Comment by blhagadorn** [comment]
> Check if the option works in Istio 1.5 if that is a possibility.
About to do that now, thanks for keeping this issue updated.

**Comment by blhagadorn** [comment]
Can confirm that this works in 1.5

**Comment by mandarjog** [comment]
@richardwxn is enabling egress working ? What is the issue here ?

**Comment by richardwxn** [comment]
@mandarjog this was already been fixed in 1.5, the reason is that the set enablement was broken for values passthrough at some point
